coordinator/points_writer_test.go

```diff
@@ -145,7 +222,7 @@ func TestPointsWriter_MapShards_Invalid(t *testing.T) {
 	}
 
 	// Add a point that goes beyond the current retention policy.
-	pr.AddPoint("cpu", 1.0, time.Now().Add(-2*time.Hour), nil)
+	pr.AddPoint("cpu", 1.0, time.Now().Add(time.Hour), nil)
```

Comment:
What's the goal of this change? One hour into the future seems like the wrong thing to test not mapping points that are older than the retention policy duration. The mocking done in this test ignore the timestamp, and the code you added doesn't. The RP returned from `NewRetentionPolicy` on line 206 will always return a ShardGroupInfo with times at unix epoch: https://github.com/influxdata/influxdb/blob/master/coordinator/points_writer_test.go#L498

The test (or helper methods) should be updated or thrown out as "testing the wrong things"